src/rust/ide/src/controller/module.rs

```diff
@@ -137,6 +161,22 @@ shared! { Handle
             Ok(())
         }
 
+        /// Returns metadata for given node.
+        pub fn node_metadata(&mut self, id:ast::ID) -> FallibleResult<NodeMetadata> {
+            self.module.metadata.ide.get(&id).cloned()
+                .ok_or_else(|| NodeMetadataNotFound(id).into())
+        }
+
+        /// Modify metadata of given node.
```

Comment:
please also describe how it behaves when there is no node metadata present for the given node